panic: runtime error

All my pipelines started to fail with this error: `panic: runtime error: invalid memory address or nil pointer dereference`

These are the steps in the pipeline:
```yaml
      - name: cdk diff
        run: cdk diff --progress=events &> >(tee cdk.log)
        working-directory: cdk

      - name: Save diff in PR
        run: >
          cdk-notifier
          --owner             ${{ github.repository_owner }}
          --repo              ${{ github.event.repository.name }}
          --token             ${{ github.token }}
          --log-file          ./cdk.log
          --pull-request-id   ${{ github.event.pull_request.number  }}
          --tag-id "all stacks - ${{github.base_ref}} "
        working-directory: cdk
```

<details>
<summary>cdk diff</summary>

Warning:  aws-cdk-lib.CfnResource#addDependsOn is deprecated.
  use addDependency
  This API will be removed in the next major release.
Warning:  aws-cdk-lib.aws_stepfunctions.TaskStateBaseProps#timeout is deprecated.
  use `taskTimeout`
  This API will be removed in the next major release.
Warning:  aws-cdk-lib.aws_stepfunctions.TaskStateBaseProps#timeout is deprecated.
  use `taskTimeout`
  This API will be removed in the next major release.
Warning:  aws-cdk-lib.aws_stepfunctions.TaskStateBaseProps#timeout is deprecated.
  use `taskTimeout`
  This API will be removed in the next major release.
Warning:  aws-cdk-lib.aws_stepfunctions.TaskStateBaseProps#timeout is deprecated.
  use `taskTimeout`
  This API will be removed in the next major release.

Stack dev-ss
There were no differences
Stack dev-contactRegistration
Resources
[~] AWS::Lambda::Function LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8a LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aFD4BFC8A 
 ├─ [~] Code
 │   └─ [~] .S3Key:
 │       ├─ [-] eb5b005c858404ea0c8f68098ed5dcdf5340e024[61](https://github.com/pcfco/platform/actions/runs/4219499844/jobs/7358206286#step:8:62)f149751d10f59c210d5ef8.zip
 │       └─ [+] e45ee2082d227db1b6f0292696ce5ce2b061c105d15efb341925ca040d1feb[68](https://github.com/pcfco/platform/actions/runs/4219499844/jobs/7358206286#step:8:69).zip
 └─ [~] Metadata
     └─ [~] .aws:asset:path:
         ├─ [-] asset.eb5b005c858404ea0c8f68098ed5dcdf5340e02461f149751d10f59c210d5ef8
         └─ [+] asset.e45ee2082d227db1b6f0292[69](https://github.com/pcfco/platform/actions/runs/4219499844/jobs/7358206286#step:8:70)6ce5ce2b061c105d15efb341925ca040d1feb68

Stack dev-apps
Resources
[~] AWS::Lambda::Function LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8a LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aFD4BFC8A 
 ├─ [~] Code
 │   └─ [~] .S3Key:
 │       ├─ [-] eb5b005c858404ea0c8f68098ed5dcdf5340e02461f149[75](https://github.com/pcfco/platform/actions/runs/4219499844/jobs/7358206286#step:8:76)1d10f59c210d5ef8.zip
 │       └─ [+] e45ee2082d227db1b6f0292696ce5ce2b061c105d15efb341925ca040d1feb68.zip
 └─ [~] Metadata
     └─ [~] .aws:asset:path:
         ├─ [-] asset.eb5b005c858404ea0c8f6[80](https://github.com/pcfco/platform/actions/runs/4219499844/jobs/7358206286#step:8:81)98ed5dcdf5340e02461f149751d10f59c210d5ef8
         └─ [+] asset.e45ee20[82](https://github.com/pcfco/platform/actions/runs/4219499844/jobs/7358206286#step:8:83)d227db1b6f02[92](https://github.com/pcfco/platform/actions/runs/4219499844/jobs/7358206286#step:8:93)6[96](https://github.com/pcfco/platform/actions/runs/4219499844/jobs/7358206286#step:8:97)ce5ce2b061c[105](https://github.com/pcfco/platform/actions/runs/4219499844/jobs/7358206286#step:8:106)d15efb341925ca040d1feb68

Stack dev-monitoring
There were no differences

</details>


<details>
<summary>Full Error log</summary>
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x8fbdaa]
goroutine 1 [running]:
github.com/karlderkaefer/cdk-notifier/provider.transform(...)
	/home/circleci/project/provider/github.go:50
github.com/karlderkaefer/cdk-notifier/provider.(*GithubClient).CreateComment(0xc0000ae0e0)
	/home/circleci/project/provider/github.go:64 +0xaa
github.com/karlderkaefer/cdk-notifier/provider.postComment({0xbb6c90, 0xc0000ae0e0}, {{0x7ffe36fbd7be, 0x9}, {0x7ffe36fbd7e7, 0x11}, {0x7ffe36fbd779, 0x8}, {0x7ffe36fbd76c, 0x5}, ...})
	/home/circleci/project/provider/api.go:102 +0x3de
github.com/karlderkaefer/cdk-notifier/provider.(*GithubClient).PostComment(0x7ffe36fbd7be?)
	/home/circleci/project/provider/github.go:82 +0xa5
github.com/karlderkaefer/cdk-notifier/cmd.glob..func1(0xf[29](https://github.com/pcfco/platform/actions/runs/4219499844/jobs/7358206286#step:9:30)560, {0xada7e9?, 0xc?, 0xc?})
	/home/circleci/project/cmd/root.go:53 +0x[34](https://github.com/pcfco/platform/actions/runs/4219499844/jobs/7358206286#step:9:35)a
github.com/spf13/cobra.(*Command).execute(0xf29560, {0xc000024[35](https://github.com/pcfco/platform/actions/runs/4219499844/jobs/7358206286#step:9:36)0, 0xc, 0xc})
	/home/circleci/go/pkg/mod/github.com/spf13/cobra@v1.6.1/command.go:920 +0x847
github.com/spf13/cobra.(*Command).ExecuteC(0xf29560)
	/home/circleci/go/pkg/mod/github.com/spf13/cobra@v1.6.1/command.go:1044 +0x3bd
github.com/spf13/cobra.(*Command).Execute(...)
	/home/circleci/go/pkg/mod/github.com/spf13/cobra@v1.6.1/command.go:968
github.com/karlderkaefer/cdk-notifier/cmd.Execute()
	/home/circleci/project/cmd/root.go:70 +0x25
main.main()
	/home/circleci/project/main.go:6 +0x17
Error: Process completed with exit code 2.

</details>

--- a/README.md
+++ b/README.md
@@ -87,6 +87,9 @@ This is an example how the diff would like on github
 cdk-notifier -l data/cdk-small.log -t test
 ```
 
+Please note: GitLab, Bitbucket or Github tokens needs permission to write comments to PR.
+As an example for GitHub PAT it requires `pull requests: read and write`.
+
 ![](images/diff.png)
 
 ## Support for CI Systems

--- a/provider/github.go
+++ b/provider/github.go
@@ -2,6 +2,8 @@ package provider
 
 import (
 	"context"
+	"errors"
+
 	"github.com/google/go-github/v37/github"
 	"github.com/karlderkaefer/cdk-notifier/config"
 	"golang.org/x/oauth2"
@@ -61,11 +63,23 @@ func transform(i *github.IssueComment) *Comment {
 
 func (gc *GithubClient) CreateComment() (*Comment, error) {
 	comment, _, err := gc.Issues.CreateComment(gc.Context, gc.Config.RepoOwner, gc.Config.RepoName, gc.Config.PullRequestID, &github.IssueComment{Body: &gc.CommentContent})
+	if err != nil {
+		return nil, err
+	}
+	if comment == nil {
+		return nil, errors.New("Comment is nil. Please check your GitHub token.")
+	}
 	return transform(comment), err
 }
 
 func (gc *GithubClient) UpdateComment(id int64) (*Comment, error) {
 	editedComment, _, err := gc.Issues.EditComment(gc.Context, gc.Config.RepoOwner, gc.Config.RepoName, id, &github.IssueComment{Body: &gc.CommentContent})
+	if err != nil {
+		return nil, err
+	}
+	if editedComment == nil {
+		return nil, errors.New("Comment is nil. Please check your GitHub token.")
+	}
 	return transform(editedComment), err
 }
 

